src/data/battler-tags.ts

```diff
@@ -885,8 +885,9 @@ export class PowderTag extends BattlerTag {
       const movePhase = pokemon.scene.getCurrentPhase();
       if (movePhase instanceof MovePhase) {
         const move = movePhase.move.getMove();
-        if (pokemon.getMoveType(move) === Type.FIRE) {
-          movePhase.cancel();
+        const weather = pokemon.scene.arena.weather;
+        if (pokemon.getMoveType(move) === Type.FIRE && !(weather && weather.weatherType === WeatherType.HEAVY_RAIN && !weather.isEffectSuppressed(pokemon.scene))) {
+          movePhase.fail();
```

Comment:
```suggestion
          movePhase.fail();
          movePhase.showMoveText();
```

"{Pokemon} used {move}!" is meant to show before Powder explodes but currently doesn't. This is kind of a bandaid fix since it's awkward to implement this within the `PRE_MOVE` or `MOVE` lapse types